health/health_log.c

```diff
@@ -185,23 +208,27 @@ inline ssize_t health_alarm_log_read(RRDHOST *host, FILE *fp, const char *filena
                 ae = callocz(1, sizeof(ALARM_ENTRY));
             }
             else if(unlikely(*pointers[0] == 'U')) {
-                // find the original
-                for(ae = host->health_log.alarms; ae; ae = ae->next) {
-                    if(unlikely(unique_id == ae->unique_id)) {
-                        if(unlikely(*pointers[0] == 'A')) {
-                            error("HEALTH [%s]: line %zu of file '%s' adds duplicate alarm log entry %u. Using the later."
-                                  , host->hostname, line, filename, unique_id);
-                            *pointers[0] = 'U';
-                            duplicate++;
+                if(likely(!alarm_isrepeating(host, alarm_id))) {
```

Comment:
I also don't get this part. This `if` seems to have been put here because the originial contributor didn't understand what this case does. We can't just ignore a duplicate entry when the alarm is repeating.